src/core/Akka/IO/SocketEventArgsPool.cs

```diff
@@ -132,7 +101,8 @@ public static void SetBuffer(this SocketAsyncEventArgs args, IEnumerable<ByteStr
             else
             {
                 args.SetBuffer(null, 0, 0);
-                args.BufferList = dataCollection.SelectMany(d => d.Buffers).ToList();
+                // TODO: fix this before we ship
+                args.BufferList = new List<ArraySegment<byte>>(dataCollection.Select(bs => new ArraySegment<byte>(bs.ToArray())));
```

Comment:
Rather than go down that road, I'd prefer to rewrite Akka.IO to not depend on `SocketAsyncEventArgs` and all of the complicated junk inside the TCP and UDP Akka.IO actors. I think some major simplification is in-order there for .NET 6 at least.